mlflow/store/artifact/http_artifact_repo.py

```diff
@@ -1,32 +1,28 @@
 import os
-import requests
 import posixpath
 
 from mlflow.entities import FileInfo
 from mlflow.store.artifact.artifact_repo import ArtifactRepository, verify_artifact_path
+from mlflow.tracking._tracking_service.utils import _get_default_host_creds
 from mlflow.utils.file_utils import relative_path_to_artifact_path
-from mlflow.utils.rest_utils import augmented_raise_for_status
+from mlflow.utils.rest_utils import augmented_raise_for_status, http_request
 
 
 class HttpArtifactRepository(ArtifactRepository):
     """Stores artifacts in a remote artifact storage using HTTP requests"""
 
     def __init__(self, artifact_uri):
         super().__init__(artifact_uri)
-        self._session = requests.Session()
-
-    def __del__(self):
-        if hasattr(self, "_session"):
-            self._session.close()
+        self._host_creds = _get_default_host_creds(artifact_uri)
```

Comment:
On second thought, can we fetch new host creds each time `log_artifact()` / `download_artifact()` is called? This allows users to apply auth changes throughout the lifespan of the artifact repo.

Reply:
Sure, if there is a use for that 👍